grails-app/services/org/pih/warehouse/inventory/InventoryItemDataService.groovy

```diff
@@ -1,15 +1,15 @@
 package org.pih.warehouse.inventory
 
-import grails.gorm.services.Join
+import grails.gorm.services.Query
 import grails.gorm.services.Service
 
 @Service(InventoryItem)
 interface InventoryItemDataService {
 
     InventoryItem save(InventoryItem inventoryItem);
 
-    @Join("product")
-    InventoryItem get(String id);
+    @Query("select item from InventoryItem item join fetch item.product where item.id = $id")
```

Comment:
couldn't you make it work with `@Join` annotation? 
I was hoping it'd work, because in the resource I've sent, there was an example of `findAnother` and was wondering what might hide under the `Another`. 
Did you check the same, but not with `get`, but with `find`? 
You could compare the classes that are generated with each of them.
I would be grateful for the summary of your tries.

Reply:
@alannadolny could you try this?
```groovy
@Join("product")
InventoryItem findWithProduct(String id)
```

Reply:
I tried this, but I didn't restart the application and it didn't work without it 🙈 🙉

Reply:
@alannadolny cool, sorry for nitpickness, but if `getWithProduct` would also work, I would prefer this version instead of `find`.